subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -7872,29 +7872,63 @@ void ull_dle_max_time_get(struct ll_conn *conn, uint16_t *max_rx_time,
 	return dle_max_time_get(conn, max_rx_time, max_tx_time);
 }
 
+static bool dle_remote_valid(struct ll_conn *conn)
+{
+	const struct data_pdu_length *dle = &conn->lll.dle.remote;
+
+	if (!IN_RANGE(dle->max_rx_octets, PDU_DC_PAYLOAD_SIZE_MIN, PDU_DC_PAYLOAD_SIZE_MAX)) {
+		return false;
+	}
+
+	if (!IN_RANGE(dle->max_tx_octets, PDU_DC_PAYLOAD_SIZE_MIN, PDU_DC_PAYLOAD_SIZE_MAX)) {
+		return false;
+	}
+
+#if defined(CONFIG_BT_CTLR_PHY)
+	if (!IN_RANGE(dle->max_rx_time, PDU_DC_PAYLOAD_TIME_MIN, PDU_DC_PAYLOAD_TIME_MAX_CODED)) {
+		return false;
+	}
+
+	if (!IN_RANGE(dle->max_tx_time, PDU_DC_PAYLOAD_TIME_MIN, PDU_DC_PAYLOAD_TIME_MAX_CODED)) {
+		return false;
+	}
+#else
+	if (!IN_RANGE(dle->max_rx_time, PDU_DC_PAYLOAD_TIME_MIN, PDU_DC_PAYLOAD_TIME_MAX)) {
+		return false;
+	}
+
+	if (!IN_RANGE(dle->max_tx_time, PDU_DC_PAYLOAD_TIME_MIN, PDU_DC_PAYLOAD_TIME_MAX)) {
+		return false;
+	}
+#endif /* CONFIG_BT_CTLR_PHY */
+
+	return true;
+}
+
 uint8_t ull_dle_update_eff(struct ll_conn *conn)
 {
 	uint8_t dle_changed = 0;
+	bool remote_valid = dle_remote_valid(conn);
 
-	const uint16_t eff_tx_octets =
+	const uint16_t eff_tx_octets = remote_valid ?
 		MAX(MIN(conn->lll.dle.local.max_tx_octets, conn->lll.dle.remote.max_rx_octets),
-		    PDU_DC_PAYLOAD_SIZE_MIN);
-	const uint16_t eff_rx_octets =
+		    PDU_DC_PAYLOAD_SIZE_MIN) : conn->lll.dle.local.max_tx_octets;
```

Comment:
We should use eff.max... values here and not local.max...., right? Otherwise the local.max... values will over-rule the ones previously suggested by peer, and that could very well be a bad idea?